java/core/src/main/java/co/worklytics/psoxy/gateway/impl/CommonRequestHandler.java

```diff
@@ -147,30 +149,14 @@ public HttpEventResponse handle(HttpEventRequest request) {
                     .build();
         }
 
-
         Optional<HttpEventResponse> healthCheckResponse = healthCheckRequestHandler.handleIfHealthCheck(request);
         if (healthCheckResponse.isPresent()) {
             return healthCheckResponse.get();
         }
 
-        // parse requested URL, re-writing host/etc
-        String requestedTargetUrl = parseRequestedTarget(request);
-
-        // if requested target URL has tokenized components, reverse
-        String clearTargetUrl = reverseTokenizedUrlComponents(requestedTargetUrl);
-
-        boolean tokenizedURLReversed = ObjectUtils.notEqual(requestedTargetUrl, clearTargetUrl);
-
-        // Using original URL to check sanitized rules, as they should match the original URL. It could contain tokenized components.
-        // Examples:
-        // /v1/accounts/p~12adsfasdfasdf31
-        // /v1/accounts/12345
-        URL originalRequestedURL = new URL(requestedTargetUrl);
-        // And the URL to use for source request; it could contain the reversed tokenized components
-        URL targetForSourceApiRequest = new URL(clearTargetUrl);
-
+        RequestUrls requestUrls = buildRequestedUrls(request);
```

Comment:
I'd wrap from here the try-catch that starts in L201

Reply:
added more